clusterman/autoscaler/pool_manager.py

```diff
@@ -247,6 +233,44 @@ def _reload_resource_groups(self) -> None:
         self.resource_groups = resource_groups
         logger.info(f'Loaded resource groups: {list(resource_groups)}')
 
+    def _filter_scale_up_options_for_pod(
+        self,
+        pod: KubernetesPod,
+        scale_up_options: Mapping[str, List[ClusterNodeMetadata]],
+    ) -> Mapping[str, List[ClusterNodeMetadata]]:
+        filtered_options: Mapping[str, List[ClusterNodeMetadata]] = defaultdict(list)
+        for group_id, options in scale_up_options.items():
+            for option in options:
+                try:
+                    if option.agent.unschedulable:
+                        raise _FilterException('node is not schedulable')
+
+                    if total_pod_resources(pod) > option.agent.allocated_resources:
+                        raise _FilterException('insufficient resources')
+
+                    if option.agent.max_tasks and option.agent.task_count >= option.agent.max_tasks:
+                        raise _FilterException('node running max tasks')
+
+                    for key, value in (pod.spec.node_selector or {}).items():
+                        if option.agent.labels.get(key) != value:
+                            raise _FilterException('mismatched node selector')
```

Comment:
I think you should do a set diff between the node selectors and agent labels to  get all labels that aren't on the node, and then output that for easier debugging.